doc/overlays.xml

```diff
@@ -96,4 +96,68 @@ overridden and/or new packages.</para>
 
 </section>
 
+<section xml:id="sec-overlays-extend-modify">
+<title>Overlays used to extend or modify pkgs</title>
+
+<para>
+If you want to extend nixpkgs with your own packages: add new packages, alter existing ones then you can use this code <filename>nixcloud-pkgs.nix</filename>:
+</para>
+
+<programlisting>
+{  pkgs ? import &lt;nixpkgs> {}
+,  nixcloud-backend   ? pkgs.stdenv.lib.cleanSource ../nixcloud-backend
+,  nixcloud-frontend  ? pkgs.stdenv.lib.cleanSource ../nixcloud-frontend
+,  nixcloud-editor    ? pkgs.stdenv.lib.cleanSource ../nixcloud-editor
+, ...
+}:
+let
+  nc-backend = nixcloud-backend;
+  nc-frontend = nixcloud-frontend;
+  
+  newpkgs = import pkgs.path { overlays = [ (pkgsself: pkgssuper: {
+    nixcloud-backend  = pkgs.callPackage nc-backend { inherit newpkgs; };
+    nixcloud-frontend = pkgs.callPackage "${nc-frontend}/release.nix" {
+      inherit nixcloud-editor;
+    };
+  } ) ]; };
+in newpkgs
+</programlisting>
+
+<para>
+If you want to install a package from this new set, do this:
+
+<programlisting>
+nix-build nixcloud-pkgs.nix -A nixcloud-backend
+</programlisting>
+
+If you create additional services, with the nixos module system, then you can reference your new set of modules like this:
+
+<programlisting>
+{...}:
+let
+  ncpkgs = import ./nixcloud-pkgs.nix {};
```

Comment:
If I add overlays to global `nixpkgs.overlays`, they are not available by in my user environment. Is this intended?